src/workerd/io/actor-sqlite.c++

```diff
@@ -406,27 +433,17 @@ kj::Maybe<kj::Promise<void>> ActorSqlite::setAlarm(
     kj::Maybe<kj::Date> newAlarmTime, WriteOptions options) {
   requireNotBroken();
 
-  // TODO(soon): Need special logic to handle case where actor is using alarms without using
-  // other storage?
+  // TODO(someday): When deleting alarm data in an otherwise empty database, clear the database to
+  // free up resources?
 
-  if (!haveDeferredDelete) {
-    // Skip setting alarm if alarm is already set to the given value.
-    //
-    // If we're in the alarm handler and haven't set the time yet, we can't perform this
-    // optimization as the current alarm time may be equal to the requested time but we indicate
-    // to the actor in getAlarm() that there is no alarm set, therefore we need to act like that
-    // in setAlarm().
-    //
-    // After the first write in the handler occurs, the logic here is correct again as the current
-    // alarm time would match what we are reporting to the user from getAlarm().
-    //
-    // TODO(now): optimization no longer useful in sqlite?
```

Comment:
Is the idea that this is no longer useful because sqlite won't actually rewrite the page if it's unchanged? Or if it's something else, why?

Reply:
I'd added an optimization in the SqliteMetadata caching layer to skip writes if the write already matched the cached value, so yeah, writing the same value wouldn't currently even reach sqlite, here.

Also, I think the motivation for the analogous comment in ActorCache is because we don't want to skip the alarm dirtiness logic when the current alarm time matches the requested alarm time but we still have a deferred delete.  But in the case of the below code, we're running the dirtiness logic regardless (even though I guess we'd similarly only need to run it when we have a deferred delete, assuming handlers and transactions nest sanely).